sorting/bubble_sort_optimized.cpp

```diff
@@ -0,0 +1,62 @@
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+#include <cassert>   /// for assert
+
+namespace sorting {
+
+    /**
+     * @brief Sorts the given vector using the bubble sort algorithm.
+     * Optimized by reducing the number of passes in average cases.
+     * @param nums the vector of integers to be sorted
+     * @return reference to the sorted vector
+     */
+    std::vector<int>& bubble_sort(std::vector<int>& nums) {
+        int flag=0;
```

Comment:
Shouldn't flag be a bool?
```suggestion
        bool flag =  false;
```